### PR TITLE
Support tls_disabled_versions for dynflow core

### PR DESCRIPTION
**Pull request by stbenjam**

### CODE REVIEW

File: manifests/plugin/dynflow.pp [hunk 1]
```diff
@@ -4,21 +4,23 @@
...
-# $core_listen::     Address to listen on for the dynflow core service
+# $core_listen::           Address to listen on for the dynflow core service
 #
-# $core_port::       Port to use for the local dynflow core service
+# $core_port::             Port to use for the local dynflow core service
 #
-# $ssl_disabled_ciphers:: Disable SSL ciphers
+# $ssl_disabled_ciphers::  Disable SSL ciphers
+#
+# $tls_disabled_versions:: Disable TLS versions
```
Comment (ekohl) [inline manifests/plugin/dynflow.pp]: Some examples could use useful if we allow any string.

**Comment by stbenjam** [comment]
Needs https://github.com/theforeman/smart_proxy_dynflow/pull/43 first

File: manifests/plugin/dynflow.pp [hunk 2]
```diff
@@ -28,6 +30,7 @@
   String $core_listen = $::foreman_proxy::plugin::dynflow::params::core_listen,
   Integer[0, 65535] $core_port = $::foreman_proxy::plugin::dynflow::params::core_port,
   Optional[Array[String]] $ssl_disabled_ciphers = $::foreman_proxy::plugin::dynflow::params::ssl_disabled_ciphers,
+  Optional[Array[String]] $tls_disabled_versions = $::foreman_proxy::plugin::dynflow::params::tls_disabled_versions,
```
Comment (sean797) [inline manifests/plugin/dynflow.pp]: Should this be `Float` instead of `String`?

Comment (stbenjam) [reply manifests/plugin/dynflow.pp]: I'd not really consider a version to be a float

Comment (ekohl) [reply manifests/plugin/dynflow.pp]: If anything I'd make it ```Optional[Array[Enum['1.1', '1.2']]]``` since 1.0 is always disabled anyway.

Comment (stbenjam) [reply manifests/plugin/dynflow.pp]: 1.3 is coming, too.   I can make it an enum or float if you really want, but if a user gives bad data it'll get logged and nothing bad will really happen.

Comment (ekohl) [reply manifests/plugin/dynflow.pp]: That's why I accepted String already :)

Comment (sean797) [reply manifests/plugin/dynflow.pp]: Okay, lets stick with `String` 👍

**Comment by ekohl** [comment]
Merging now that the dynflow change is merged. We can always improve docs in a patch version.

**Comment by ekohl** [comment]
Looks like I was a bit too eager: https://github.com/theforeman/puppet-foreman_proxy/pull/408 should fix it.

**Pull request merged**